### ISSUE TITLE
matrix.release() throws OpenCV error

### ISSUE THREAD

**Issue by NiveditN** [peterbraden/node-opencv#288]
This is in context of handling memory leak in camera.read() by using your newly included matrix.release() function on Raspberry Pi 2.

This is my code: 

~~~~
camera.read(function(err, im) {
  if (err) throw err;

  im.detectObject('./node_modules/opencv/data/haarcascade_frontalface_alt2.xml', {}, function(err, faces) {
    if (err) throw err;

    for (var i = 0; i < faces.length; i++) {
      face = faces[i];
      im.rectangle([face.x, face.y], [face.width, face.height], rectColor, rectThickness);
    }

    socket.emit('frame', { buffer: im.toBuffer() });
    im.release()
  });
// im.release() -- Position 2
});
~~~~

But upon execution, it runs for a few seconds then fails with a Segmentation fault.

I moved im.release() to Position 2 as shown in the code, but then it fails with the following error:

~~~~
OpenCV Error: Bad flag (parameter or structure field) (Unrecognized or unsupported array type) in cvGetMat, file /home/pi/installers/opencv-2.4.10/modules/core/src/array.cpp, line 2482
[ Matrix 240x320 ]
/home/pi/projects/face-detection-node-opencv/server/lib/routes/socket.js:26
        if (err) throw err;
                       ^
Error: /home/pi/installers/opencv-2.4.10/modules/core/src/array.cpp:2482: error: (-206) Unrecognized or unsupported array type in function cvGetMat

    at Error (native)
~~~~

What is going wrong? Is the matrix 'im' not getting released as it should using release function?

**Comment by peterbraden** [comment]
I think this was fixed in ddf0313d3a67597b30b991bd52bfeb3baac75f85 - can you re-install and try again?

**Comment by danschultzer** [comment]
I'll close this issue for now, since there hasn't been any response for a year.
